Makefile

```diff
@@ -190,27 +190,26 @@ unit-sup: build-launcher-for-supervisor-tests
 .PHONY: build-launcher-for-supervisor-tests
 
 # Lints we need to work through and decide as a team whether to allow or fix
-UNEXAMINED_LINTS = clippy::cyclomatic_complexity \
-                   clippy::large_enum_variant \
-                   clippy::len_without_is_empty \
-                   clippy::module_inception \
-                   clippy::needless_pass_by_value \
-                   clippy::needless_return \
-                   clippy::new_ret_no_self \
-                   clippy::new_without_default \
-                   clippy::new_without_default_derive \
-                   clippy::question_mark \
-                   clippy::redundant_field_names \
-                   clippy::too_many_arguments \
-                   clippy::trivially_copy_pass_by_ref \
-                   clippy::wrong_self_convention \
-                   renamed_and_removed_lints
+UNEXAMINED_LINTS =
 
 # Lints we disagree with and choose to keep in our code with no warning
-ALLOWED_LINTS =
+ALLOWED_LINTS = clippy::module_inception \
+                clippy::needless_return \
+                clippy::new_ret_no_self \
+                clippy::new_without_default \
+                clippy::new_without_default_derive \
```

Comment:
https://rust-lang-nursery.github.io/rust-clippy/master/index.html#new_without_default_derive

Basically the same as above, but with types that could have their `Default` implementation derived.

[new_without_default_derive.txt](https://github.com/habitat-sh/habitat/files/2813946/new_without_default_derive.txt)

Reply:
These three(`new_ret_no_self`, `new_without_default` and `new_without_default_derive`) seem like they should go together. If we use one then we should use them all.